docs/how-to/create_plugin.rst

```diff
@@ -0,0 +1,157 @@
+.. _how_to_create_plugin:
+
+How to create a plugin
+======================
+
+Plugins help to bring new tools and programming languages into Craft Parts.
+This document contains instructions on how to recreate the simple
+:py:class:`~craft-parts.plugins.dump_plugin.DumpPlugin` plugin.
```

Comment:
Is this supposed to link to something? It just renders as monospaced text for me.

Reply:
No, that's just how I saw other documentation format a python class name. I can make it link to the dump plugin's documentation if you think that's better, though!

Reply:
> I can make it link to the dump plugin's documentation if you think that's better, though!

yes please